### PR TITLE
Generate deltas

### LINKED ISSUE #51: create delta packages on Windows
Following on from #47, we need to enable delta package creation _and_ publishing:
- [electron-windows-installer](https://github.com/electron/windows-installer) has a `remoteReleases` parameter which it should use to retrieve existing releases
- we'd need to then upload the right file as part of the release (see #47)
- there's probably some other work in the app to verify we're applying these updates

### PR DESCRIPTION
**Pull request by joshaber**
Fixes #51 

Some notable notes about the changes here:

1. We bake the update channel (e.g., `beta` or `production`) into the app at build time.
2. The Windows exe name is now `GitHubWindows` (no space). This is because Squirrel.Windows can't diff files with spaces in their names, so it'd always fall back to downloading the full update. Our shortcut is still created as `GitHub Desktop`. The exe name change has an unfortunate side effect: the first update where we make this change, "restart now" won't work as Squirrel will still be looking for the old exe name. I think that's OK given that it's a one-time problem for a beta.
3. I tried updating `electron-winstaller` to get the GIF RAM usage fix, but there's a bug in Squirrel.Windows with creating diffs for some of our files (https://github.com/Squirrel/Squirrel.Windows/issues/1101).

#### Testing

I tested this using the new beta channel:

1. Created a new beta release.
1. Verified the delta is uploaded to Central and is smaller than the full release.
1. Installed the beta build.
1. Created another beta release.
1. Verified the delta is uploaded to Central and is smaller than the full release.
1. Check for updates with the installed beta.
1. Verify that the installed beta updates to the new beta using the delta, via the logging in `SquirrelSetup.log`.

### CODE REVIEW

File: app/package.json [hunk 1]
```diff
@@ -3,13 +3,13 @@
...
   "companyName": "GitHub, Inc.",
-  "version": "0.6.2",
+  "version": "0.6.3-beta6",
   "main": "./main.js",
   "repository": {
     "type": "git",
     "url": "https://github.com/desktop/desktop.git"
   },
-  "description": "Extend your GitHub workflow beyond your browser with GitHub Desktop",
+  "description": "Simple collaboration from your desktop",
```
Comment (joshaber) [inline app/package.json]: This is the same byline we're using elsewhere and better fits our goals as discussed at the mini summit.

**Comment by joshaber** [comment]
This is ready for review 🙏

File: changelog.json [hunk 2]
```diff
@@ -17,6 +17,14 @@
...
       "[Fixed] Move the 'Forgot Password' link to fix the tab order of the sign in view - #2200"
     ],
+    "0.6.3-beta7": [],
+    "0.6.3-beta6": [],
+    "0.6.3-beta5": [],
+    "0.6.3-beta4": [],
+    "0.6.3-beta3": [],
+    "0.6.3-beta2": [],
+    "0.6.3-beta1": [],
+    "0.6.3-beta0": [],
```
Comment (niik) [inline changelog.json]: Should we keep these? Will they be rendered on desktop.github.com?

Comment (joshaber) [inline changelog.json]: I think we should keep them. Those _are_ real beta releases. They won't be shown on the site as that only shows releases from the `production` channel.

**Comment by joshaber** [comment]
giphy

**Pull request merged**